test/test_filter.py

```diff
@@ -0,0 +1,83 @@
+__copyright__ = """
+Copyright (C) 2020 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+import math
+import pytest
+import numpy as np
+
+from grudge.eager import EagerDGDiscretization
+from pyopencl.tools import (  # noqa
+    pytest_generate_tests_for_pyopencl as pytest_generate_tests,
+)
+from mirgecom.filter import get_spectral_filter
+
+
+@pytest.mark.parametrize("cutoff", [3, 4, 8])
+@pytest.mark.parametrize("dim", [1, 2, 3])
+@pytest.mark.parametrize("order", [1, 2, 3, 4])
+def test_filter_coeff(ctx_factory, cutoff, dim, order):
+    """
+    Tests that the filter coefficients have the right shape
+    and a quick sanity check on the values at the filter
+    band limits.
+    """
+    cl_ctx = ctx_factory()
+    nel_1d = 16
```

Comment:
I seem to recall discussing that doing this on more than one element did not yield an effective test of the filter (because element was effectively linear). Why doesn't this test still suffer from that problem?

Reply:
Please stand-by.  I have not yet checked in the complete filter or fixes we discussed.

Reply:
This is being addressed in its own PR #27